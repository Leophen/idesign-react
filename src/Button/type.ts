export interface ButtonProps {
  /**
   * 是否聚焦状态
   * @default false
   */
  active?: boolean;
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 是否禁用按钮
   * @default false
   */
  disabled?: boolean;
  /**
   * 按钮形状
   * @default round
   */
  shape?: 'square' | 'round' | 'circle';
  /**
   * 按钮尺寸
   * @default medium
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 按钮类型，用于描述组件不同的应用场景
   * @default primary
   */
  type?: 'info' | 'primary' | 'error' | 'warning' | 'success';
  /**
   * 按钮形式
   * @default base
   */
  variant?: 'base' | 'outline' | 'dashed' | 'text';
  /**
   * 内置图标
   */
  icon?: string;
  /**
   * 点击按钮触发事件
   */
  onClick?: React.MouseEventHandler;
}
