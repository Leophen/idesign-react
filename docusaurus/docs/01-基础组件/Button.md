---
title: Button 按钮
---

常用的操作按钮。

## 引入

```js
import {Button} from 'idesign-react';
```

## 基本用法

基础按钮包括不同颜色状态下的填充按钮、描边按钮、虚框按钮和文字按钮。

```jsx live
<div className="demo-block-column">
  {/* 灰色 info */}
  <div className="demo-block-row">
    <Button type="info">填充按钮</Button>
    <Button type="info" variant="outline">描边按钮</Button>
    <Button type="info" variant="dashed">虚框按钮</Button>
    <Button type="info" variant="text">文字按钮</Button>
  </div>
  {/* 蓝色 primary（默认） */}
  <div className="demo-block-row">
    <Button type="primary">填充按钮</Button>
    <Button type="primary" variant="outline">描边按钮</Button>
    <Button type="primary" variant="dashed">虚框按钮</Button>
    <Button type="primary" variant="text">文字按钮</Button>
  </div>
  {/* 绿色 success */}
  <div className="demo-block-row">
    <Button type="success">填充按钮</Button>
    <Button type="success" variant="outline">描边按钮</Button>
    <Button type="success" variant="dashed">虚框按钮</Button>
    <Button type="success" variant="text">文字按钮</Button>
  </div>
  {/* 黄色 warning */}
  <div className="demo-block-row">
    <Button type="warning">填充按钮</Button>
    <Button type="warning" variant="outline">描边按钮</Button>
    <Button type="warning" variant="dashed">虚框按钮</Button>
    <Button type="warning" variant="text">文字按钮</Button>
  </div>
  {/* 红色 error */}
  <div className="demo-block-row">
    <Button type="error">填充按钮</Button>
    <Button type="error" variant="outline">描边按钮</Button>
    <Button type="error" variant="dashed">虚框按钮</Button>
    <Button type="error" variant="text">文字按钮</Button>
  </div>
</div>
```

## 聚焦状态

可以使用 `active` 属性来定义按钮是否被聚焦，该属性接受一个 `Boolean` 类型的值。

```jsx live
<div className="demo-block-column">
  {/* 填充按钮 */}
  <div className="demo-block-row">
    <Button type="info" active>Info</Button>
    <Button type="primary" active>Primary</Button>
    <Button type="success" active>Success</Button>
    <Button type="warning" active>Warning</Button>
    <Button type="error" active>Error</Button>
  </div>
  {/* 描边按钮 */}
  <div className="demo-block-row">
    <Button type="info" variant="outline" active>Info</Button>
    <Button type="primary" variant="outline" active>Primary</Button>
    <Button type="success" variant="outline" active>Success</Button>
    <Button type="warning" variant="outline" active>Warning</Button>
    <Button type="error" variant="outline" active>Error</Button>
  </div>
  {/* 虚框按钮 */}
  <div className="demo-block-row">
    <Button type="info" variant="dashed" active>Info</Button>
    <Button type="primary" variant="dashed" active>Primary</Button>
    <Button type="success" variant="dashed" active>Success</Button>
    <Button type="warning" variant="dashed" active>Warning</Button>
    <Button type="error" variant="dashed" active>Error</Button>
  </div>
  {/* 文字按钮 */}
  <div className="demo-block-row">
    <Button type="info" variant="text" active>Info</Button>
    <Button type="primary" variant="text" active>Primary</Button>
    <Button type="success" variant="text" active>Success</Button>
    <Button type="warning" variant="text" active>Warning</Button>
    <Button type="error" variant="text" active>Error</Button>
  </div>
</div>
```

## 禁用状态

可以使用 `disabled` 属性来定义按钮是否被禁用，该属性接受一个 `Boolean` 类型的值。

```jsx live
<div className="demo-block-column">
  {/* 填充按钮 */}
  <div className="demo-block-row">
    <Button type="info" disabled>Info</Button>
    <Button type="primary" disabled>Primary</Button>
    <Button type="success" disabled>Success</Button>
    <Button type="warning" disabled>Warning</Button>
    <Button type="error" disabled>Error</Button>
  </div>
  {/* 描边按钮 */}
  <div className="demo-block-row">
    <Button type="info" variant="outline" disabled>Info</Button>
    <Button type="primary" variant="outline" disabled>Primary</Button>
    <Button type="success" variant="outline" disabled>Success</Button>
    <Button type="warning" variant="outline" disabled>Warning</Button>
    <Button type="error" variant="outline" disabled>Error</Button>
  </div>
  {/* 虚框按钮 */}
  <div className="demo-block-row">
    <Button type="info" variant="dashed" disabled>Info</Button>
    <Button type="primary" variant="dashed" disabled>Primary</Button>
    <Button type="success" variant="dashed" disabled>Success</Button>
    <Button type="warning" variant="dashed" disabled>Warning</Button>
    <Button type="error" variant="dashed" disabled>Error</Button>
  </div>
  {/* 文字按钮 */}
  <div className="demo-block-row">
    <Button type="info" variant="text" disabled>Info</Button>
    <Button type="primary" variant="text" disabled>Primary</Button>
    <Button type="success" variant="text" disabled>Success</Button>
    <Button type="warning" variant="text" disabled>Warning</Button>
    <Button type="error" variant="text" disabled>Error</Button>
  </div>
</div>
```

## 不同尺寸

提供 `small`、`middle`（默认）、`large` 三种尺寸的按钮。

```jsx live
<div className="demo-block-column">
  {/* 小尺寸 small */}
  <div className="demo-block-row">
    <Button size="small">填充按钮</Button>
    <Button variant="outline" size="small">描边按钮</Button>
    <Button variant="dashed" size="small">虚框按钮</Button>
    <Button variant="text" size="small">文字按钮</Button>
  </div>
  {/* 中尺寸 medium（默认） */}
  <div className="demo-block-row">
    <Button size="medium">填充按钮</Button>
    <Button variant="outline" size="medium">描边按钮</Button>
    <Button variant="dashed" size="medium">虚框按钮</Button>
    <Button variant="text" size="medium">文字按钮</Button>
  </div>
  {/* 大尺寸 large */}
  <div className="demo-block-row">
    <Button size="large">填充按钮</Button>
    <Button variant="outline" size="large">描边按钮</Button>
    <Button variant="dashed" size="large">虚框按钮</Button>
    <Button variant="text" size="large">文字按钮</Button>
  </div>
</div>
```

## 不同形状

提供 `square`、`round`（默认）、`circle` 三种形状的按钮。

```jsx live
<div className="demo-block-column">
  {/* 方角 square */}
  <div className="demo-block-row">
    <Button shape="square">填充按钮</Button>
    <Button variant="outline" shape="square">描边按钮</Button>
    <Button variant="dashed" shape="square">虚框按钮</Button>
  </div>
  {/* 小圆角 round（默认） */}
  <div className="demo-block-row">
    <Button shape="round">填充按钮</Button>
    <Button variant="outline" shape="round">描边按钮</Button>
    <Button variant="dashed" shape="round">虚框按钮</Button>
  </div>
  {/* 大圆角 circle */}
  <div className="demo-block-row">
    <Button shape="circle">填充按钮</Button>
    <Button variant="outline" shape="circle">描边按钮</Button>
    <Button variant="dashed" shape="circle">虚框按钮</Button>
  </div>
</div>
```

## 图标按钮

通过 `icon` 属性控制按钮中的内置图标：

```jsx live fffx
<div className="demo-block-row">
  <Button size="small" icon="ThePlus">图标按钮</Button>
  <Button size="medium" icon="ThePlus">图标按钮</Button>
  <Button size="large" icon="ThePlus">图标按钮</Button>
</div>
```

## API

| 属性      | 说明                                 | 类型                                                      | 默认值    |
| --------- | ------------------------------------ | --------------------------------------------------------- | --------- |
| active    | 是否聚焦状态                         | `boolean`                                                 | `false`   |
| className | 类名                                 | `string`                                                  | `--`      |
| children  | 内容                                 | `ReactNode`                                               | `--`      |
| disabled  | 是否禁用按钮                         | `boolean`                                                 | `false`   |
| shape     | 按钮形状                             | `square` \| `round` \| `circle`                           | `round`   |
| size      | 按钮尺寸                             | `small` \| `medium` \| `large`                            | `medium`  |
| style     | 自定义样式                           | `CSSProperties`                                           | `--`      |
| type      | 按钮类型，用于描述组件不同的应用场景 | `info` \| `primary` \|  `error` \| `warning` \| `success` | `primary` |
| variant   | 按钮形式                             | `base` \| `outline` \| `dashed` \| `text`                 | `base`    |
| icon      | 内置图标                             | `string`                                                  | `--`      |
| onClick   | 点击按钮触发事件                     | `MouseEventHandler<Element>`                              | `--`      |
