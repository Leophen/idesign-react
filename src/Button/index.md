---
nav:
  title: 组件
  path: /components
group:
  title: 基础组件
  order: 1
order: 1
---

# Button 按钮

常用的操作按钮。

## 基础按钮

基础按钮包括不同颜色状态下的填充按钮、描边按钮、虚框按钮和文字按钮。

```tsx
import React from 'react';
import { Button } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      {/* 灰色 info */}
      <div className="idesign-demo-block-row">
        <Button type="info">填充按钮</Button>
        <Button type="info" variant="outline">
          描边按钮
        </Button>
        <Button type="info" variant="dashed">
          虚框按钮
        </Button>
        <Button type="info" variant="text">
          文字按钮
        </Button>
      </div>
      {/* 蓝色 primary（默认） */}
      <div className="idesign-demo-block-row">
        <Button type="primary">填充按钮</Button>
        <Button type="primary" variant="outline">
          描边按钮
        </Button>
        <Button type="primary" variant="dashed">
          虚框按钮
        </Button>
        <Button type="primary" variant="text">
          文字按钮
        </Button>
      </div>
      {/* 绿色 success */}
      <div className="idesign-demo-block-row">
        <Button type="success">填充按钮</Button>
        <Button type="success" variant="outline">
          描边按钮
        </Button>
        <Button type="success" variant="dashed">
          虚框按钮
        </Button>
        <Button type="success" variant="text">
          文字按钮
        </Button>
      </div>
      {/* 黄色 warning */}
      <div className="idesign-demo-block-row">
        <Button type="warning">填充按钮</Button>
        <Button type="warning" variant="outline">
          描边按钮
        </Button>
        <Button type="warning" variant="dashed">
          虚框按钮
        </Button>
        <Button type="warning" variant="text">
          文字按钮
        </Button>
      </div>
      {/* 红色 error */}
      <div className="idesign-demo-block-row">
        <Button type="error">填充按钮</Button>
        <Button type="error" variant="outline">
          描边按钮
        </Button>
        <Button type="error" variant="dashed">
          虚框按钮
        </Button>
        <Button type="error" variant="text">
          文字按钮
        </Button>
      </div>
    </div>
  );
};

export default App;
```

## 聚焦状态

可以使用 `active` 属性来定义按钮是否被聚焦，该属性接受一个 `Boolean` 类型的值。

```tsx
import React from 'react';
import { Button } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      {/* 填充按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" active>
          Info
        </Button>
        <Button type="primary" active>
          Primary
        </Button>
        <Button type="success" active>
          Success
        </Button>
        <Button type="warning" active>
          Warning
        </Button>
        <Button type="error" active>
          Error
        </Button>
      </div>
      {/* 描边按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" variant="outline" active>
          Info
        </Button>
        <Button type="primary" variant="outline" active>
          Primary
        </Button>
        <Button type="success" variant="outline" active>
          Success
        </Button>
        <Button type="warning" variant="outline" active>
          Warning
        </Button>
        <Button type="error" variant="outline" active>
          Error
        </Button>
      </div>
      {/* 虚框按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" variant="dashed" active>
          Info
        </Button>
        <Button type="primary" variant="dashed" active>
          Primary
        </Button>
        <Button type="success" variant="dashed" active>
          Success
        </Button>
        <Button type="warning" variant="dashed" active>
          Warning
        </Button>
        <Button type="error" variant="dashed" active>
          Error
        </Button>
      </div>
      {/* 文字按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" variant="text" active>
          Info
        </Button>
        <Button type="primary" variant="text" active>
          Primary
        </Button>
        <Button type="success" variant="text" active>
          Success
        </Button>
        <Button type="warning" variant="text" active>
          Warning
        </Button>
        <Button type="error" variant="text" active>
          Error
        </Button>
      </div>
    </div>
  );
};

export default App;
```

## 禁用状态

可以使用 `disabled` 属性来定义按钮是否被禁用，该属性接受一个 `Boolean` 类型的值。

```tsx
import React from 'react';
import { Button } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      {/* 填充按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" disabled>
          Info
        </Button>
        <Button type="primary" disabled>
          Primary
        </Button>
        <Button type="success" disabled>
          Success
        </Button>
        <Button type="warning" disabled>
          Warning
        </Button>
        <Button type="error" disabled>
          Error
        </Button>
      </div>
      {/* 描边按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" variant="outline" disabled>
          Info
        </Button>
        <Button type="primary" variant="outline" disabled>
          Primary
        </Button>
        <Button type="success" variant="outline" disabled>
          Success
        </Button>
        <Button type="warning" variant="outline" disabled>
          Warning
        </Button>
        <Button type="error" variant="outline" disabled>
          Error
        </Button>
      </div>
      {/* 虚框按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" variant="dashed" disabled>
          Info
        </Button>
        <Button type="primary" variant="dashed" disabled>
          Primary
        </Button>
        <Button type="success" variant="dashed" disabled>
          Success
        </Button>
        <Button type="warning" variant="dashed" disabled>
          Warning
        </Button>
        <Button type="error" variant="dashed" disabled>
          Error
        </Button>
      </div>
      {/* 文字按钮 */}
      <div className="idesign-demo-block-row">
        <Button type="info" variant="text" disabled>
          Info
        </Button>
        <Button type="primary" variant="text" disabled>
          Primary
        </Button>
        <Button type="success" variant="text" disabled>
          Success
        </Button>
        <Button type="warning" variant="text" disabled>
          Warning
        </Button>
        <Button type="error" variant="text" disabled>
          Error
        </Button>
      </div>
    </div>
  );
};

export default App;
```

## 不同尺寸

提供 `small`、`medium`（默认）、`large` 三种尺寸的按钮。

```tsx
import React from 'react';
import { Button } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      {/* 小尺寸 small */}
      <div className="idesign-demo-block-row">
        <Button size="small">填充按钮</Button>
        <Button variant="outline" size="small">
          描边按钮
        </Button>
        <Button variant="dashed" size="small">
          虚框按钮
        </Button>
        <Button variant="text" size="small">
          文字按钮
        </Button>
      </div>
      {/* 中尺寸 medium（默认） */}
      <div className="idesign-demo-block-row">
        <Button size="medium">填充按钮</Button>
        <Button variant="outline" size="medium">
          描边按钮
        </Button>
        <Button variant="dashed" size="medium">
          虚框按钮
        </Button>
        <Button variant="text" size="medium">
          文字按钮
        </Button>
      </div>
      {/* 大尺寸 large */}
      <div className="idesign-demo-block-row">
        <Button size="large">填充按钮</Button>
        <Button variant="outline" size="large">
          描边按钮
        </Button>
        <Button variant="dashed" size="large">
          虚框按钮
        </Button>
        <Button variant="text" size="large">
          文字按钮
        </Button>
      </div>
    </div>
  );
};

export default App;
```

## 不同形状

提供 `square`、`round`（默认）、`circle` 三种形状的按钮。

```tsx
import React from 'react';
import { Button } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      {/* 方角 square */}
      <div className="idesign-demo-block-row">
        <Button shape="square">填充按钮</Button>
        <Button variant="outline" shape="square">
          描边按钮
        </Button>
        <Button variant="dashed" shape="square">
          虚框按钮
        </Button>
      </div>
      {/* 小圆角 round（默认） */}
      <div className="idesign-demo-block-row">
        <Button shape="round">填充按钮</Button>
        <Button variant="outline" shape="round">
          描边按钮
        </Button>
        <Button variant="dashed" shape="round">
          虚框按钮
        </Button>
      </div>
      {/* 大圆角 circle */}
      <div className="idesign-demo-block-row">
        <Button shape="circle">填充按钮</Button>
        <Button variant="outline" shape="circle">
          描边按钮
        </Button>
        <Button variant="dashed" shape="circle">
          虚框按钮
        </Button>
      </div>
    </div>
  );
};

export default App;
```

<API />
