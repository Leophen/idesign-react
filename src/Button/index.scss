$info: #e7e7e7;
$info_hover: #dcdcdc;
$info_active: #c1c1c1;
$info_disabled: #ededed;
$info_font: rgba(0, 0, 0, 0.8);

$primary: #3a3cf5;
$primary_hover: #5d5ff3;
$primary_active: #2e31d5;
$primary_disabled: #9fa1fe;

$success: #67c23a;
$success_hover: #71d340;
$success_active: #4f9c28;
$success_disabled: #b3e19d;

$warning: #e69e31;
$warning_hover: #f5aa3a;
$warning_active: #c98a2b;
$warning_disabled: #f3d19e;

$error: #f16262;
$error_hover: #fc7b7b;
$error_active: #d35050;
$error_disabled: #fab6b6;

$null_bg: transparent;

.i-button {
  position: relative;
  z-index: 0;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  height: 32px;
  margin: 0;
  padding: 0 12px;
  color: #fff;
  font-size: 14px;
  white-space: nowrap;
  text-align: center;
  text-transform: none;
  vertical-align: middle;
  tab-size: 4;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
  transition: all 0.2s linear;
  user-select: none;

  // variant
  &.i-button--variant-dashed {
    border-style: dashed !important;
  }
  &.i-button--variant-text {
    border-color: transparent !important;
  }
  // size
  &.i-button--size-small {
    height: 24px;
    padding: 0 8px;
    font-size: 12px;
    .i-button-icon {
      margin-right: 4px;
    }
  }
  &.i-button--size-large {
    height: 40px;
    padding: 0 16px;
    font-size: 16px;
    .i-button-icon {
      margin-right: 8px;
    }
  }
  // shape
  &.i-button--shape-square {
    border-radius: 0px;
  }
  &.i-button--shape-circle {
    border-radius: 20px;
  }
  // disabled
  &.i-button-disabled {
    cursor: not-allowed;
  }
}

.i-button-icon {
  margin-right: 6px;
  color: inherit;
  font-size: inherit;
}

@mixin injectColor($color, $color_hover, $color_active, $color_disabled) {
  color: #fff;
  background: $color;
  &:hover {
    background: $color_hover;
  }
  &:active {
    background: $color_active;
  }

  &.i-button--variant-text,
  &.i-button--variant-outline,
  &.i-button--variant-dashed {
    color: $color;
    background: $null_bg;
    border: 1px solid $color;
    &:hover {
      color: $color_hover;
      background: $null_bg;
      border-color: $color_hover;
    }
    &:active {
      color: $color_active;
      background: $null_bg;
      border-color: $color_active;
    }
  }

  &.i-button-active {
    background: $color_active;
    &:hover {
      background: $color_active;
    }
    &.i-button--variant-text,
    &.i-button--variant-outline,
    &.i-button--variant-dashed {
      color: $color_active;
      background: $null_bg;
      border: 1px solid $color_active;
    }
  }

  &.i-button-disabled {
    background: $color_disabled;
    &:hover {
      background: $color_disabled;
    }

    &.i-button--variant-text,
    &.i-button--variant-outline,
    &.i-button--variant-dashed {
      color: $color_disabled;
      background: $null_bg;
      border-color: $color_disabled;
      &:hover {
        color: $color_disabled;
        border-color: $color_disabled;
      }
      &:active {
        color: $color_disabled;
        border-color: $color_disabled;
      }
    }
  }
}

.i-button--type-info {
  @include injectColor($info, $info_hover, $info_active, $info_disabled);
  color: $info_font;

  &.i-button--variant-text,
  &.i-button--variant-outline,
  &.i-button--variant-dashed {
    color: $info_font;
    &:hover {
      color: $info_font;
    }
    &:active {
      color: $info_font;
    }
  }

  &.i-button-active {
    color: $info_font;

    &.i-button--variant-text,
    &.i-button--variant-outline,
    &.i-button--variant-dashed {
      color: $info_font;
      &:hover {
        color: $info_font;
      }
    }
  }

  &.i-button-disabled {
    color: #aaaeb6;

    &.i-button--variant-text,
    &.i-button--variant-outline,
    &.i-button--variant-dashed {
      color: #c0c4cc;
      &:hover {
        color: #c0c4cc;
      }
      &:active {
        color: #c0c4cc;
      }
    }
  }
}

.i-button--type-primary {
  @include injectColor($primary, $primary_hover, $primary_active, $primary_disabled);
}

.i-button--type-success {
  @include injectColor($success, $success_hover, $success_active, $success_disabled);
}

.i-button--type-warning {
  @include injectColor($warning, $warning_hover, $warning_active, $warning_disabled);
}

.i-button--type-error {
  @include injectColor($error, $error_hover, $error_active, $error_disabled);
}
