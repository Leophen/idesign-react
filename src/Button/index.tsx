import React, { ReactNode } from 'react';
import classNames from 'classnames';
import './index.scss';

export interface ButtonProps {
  disabled: boolean;
  children: ReactNode;
  className: string;
}

/**
 * 按钮组件
 */
const Button = ({ disabled, children, className, ...buttonProps }: ButtonProps) => {
  return (
    <button className={classNames('i-button', className)} disabled={disabled} {...buttonProps}>
      <span className={`i-button__text`}>{children}</span>
    </button>
  );
};

export default Button;
