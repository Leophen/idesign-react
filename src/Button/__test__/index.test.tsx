import { render, fireEvent } from '@testing-library/react';
import Button from '../index';

describe('Button 组件测试', () => {
  it('children', () => {
    const { queryByText } = render(<Button>foo</Button>);
    expect(queryByText('foo')).toBeInTheDocument();
  });

  it('type', () => {
    const { container } = render(<Button type="success" />);
    expect(container.firstChild.classList.contains('i-button--type-success')).toBeTruthy();
  });

  it('variant', () => {
    const { container } = render(<Button variant="outline" />);
    expect(container.firstChild.classList.contains('i-button--variant-outline')).toBeTruthy();
  });

  it('active', () => {
    const { container } = render(<Button active />);
    expect(container.firstChild.classList.contains('i-button-active')).toBeTruthy();
  });

  it('disabled', () => {
    const clickFn = jest.fn();
    const { container } = render(<Button disabled onClick={clickFn} />);
    expect(container.firstChild).toBeDisabled();
    fireEvent.click(container.firstChild);
    expect(clickFn).toBeCalledTimes(0);
  });

  it('size', () => {
    const { container } = render(<Button size="small" />);
    expect(container.firstChild.classList.contains('i-button--size-small')).toBeTruthy();
  });

  it('shape', () => {
    const { container } = render(<Button shape="circle" />);
    expect(container.firstChild.classList.contains('i-button--shape-circle')).toBeTruthy();
  });

  it('icon', () => {
    const { asFragment } = render(<Button icon="ThePlus" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('onClick', () => {
    const clickFn = jest.fn();
    const { container } = render(<Button onClick={clickFn} />);
    fireEvent.click(container.firstChild);
    expect(clickFn).toBeCalledTimes(1);
  });
});

