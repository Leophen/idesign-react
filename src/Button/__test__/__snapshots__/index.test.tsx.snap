// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 组件测试 icon 1`] = `
<DocumentFragment>
  <button
    class="i-button i-button--type-primary"
  >
    <i
      class="i-icon icon-ThePlus i-button-icon"
    />
  </button>
</DocumentFragment>
`;
